### PR TITLE
refactor: migrate

### PR DESCRIPTION
**Pull request by jannyHou**
### Description

connect to https://github.com/strongloop/loopback-connector-cloudant/pull/225.

To support partitioned index in cloudant connector, I refactored some inner functions into private functions. 

The new cloudant driver supports `{partitioned: true/false}` as a property when create the index, while nano hasn't yet(and treat `partitioned` as an invalid property), that's why the index object generation functions in the couchdb connector still takes in `fields` only, instead of an object with property `fields`, `partitioned`.

#### Related issues

- connect to strongloop/loopback-connector-cloudant#225

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: lib/migrate.js [hunk 1]
```diff
@@ -142,7 +129,31 @@ function mixinMigrate(CouchDB) {
...
-   * Add an index for modelView propery to indexes got from modelDef
+    * Used in function `migrateOrUpdateIndex`.
+    * Create index with index fields.
+    * @param {Object} mo  The model configuration.
+    * @param {String} model  The model name.
+    * @param {String} name  The index name.
+    * @param {Object} indexObj The index object e.g. [{"foo": "asc"}, {"bar": "asc"}]
+    * @param {Function} cb
+  */
+  CouchDB.prototype.__createIndex = function(mo, model, name, fields, cb) {
```
Comment (bajtos) [inline lib/migrate.js]: I think we usually use a single underscore for private members, e.g. `_createIndex`. Is there any particular reason for using two underscores?

Comment (jannyHou) [reply lib/migrate.js]: good catch 🙈

**Comment by jannyHou** [comment]
@slnode test please

**Comment by rmg** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Pull request merged**